File: cases/taint/php/closure-capture-negative/closure_capture_negative.php
<?php
function dfb_source(): string { // DFB-SOURCE: closure-capture-input
    return "tainted";
}

function dfb_sink(string $value): void {} // DFB-SINK: closure-capture-sink

function makeReporter(): \Closure
{
    $tainted = dfb_source(); // DFB-WITNESS: closure-capture-bind
    $captured = "clean";
    return function () use ($captured): void {
        dfb_sink($captured);
    };
}

function run(): void {
    $reporter = makeReporter();
    $reporter();
}

File: cases/taint/scala/closure-capture-negative/ClosureCaptureNegative.scala
package dataflowbench

object ClosureCaptureNegative {
  def dfb_source(): String = { // DFB-SOURCE: closure-capture-input
    "tainted"
  }

  def dfb_sink(value: String): Unit = {} // DFB-SINK: closure-capture-sink

  def makeReporter(): () => Unit = {
    val tainted = dfb_source() // DFB-WITNESS: closure-capture-bind
    val captured = "clean"
    () => dfb_sink(captured)
  }

  def run(): Unit = {
    val reporter = makeReporter()
    reporter()
  }
}
